README.md

```diff
@@ -51,85 +51,125 @@ some kind of discriminant:
 enum Colors of Number {
   red,
   green,
-  blue
+  blue,
 }
 
 // Each auto-initialized member value is a `String` whose value is the SV of its member name.
 enum PlayState of String {
   idle,
   running,
-  paused
+  paused,
 }
 
 // Each auto-initialized member value is a `Symbol` whose description is the SV of its member name.
 enum Symbols of Symbol {
   alpha,
-  beta
+  beta,
 }
 
-enum Named {
-  identifierName,
-  "string name",
-  [expr]
+// A special `Enum.ADT` mapper exists that can be used to define algebraic data types:
+enum Message of Enum.ADT {
+  Move{x, y},
+  Write(text),
+  Quit
 }
 
-// Accessing enum values:
-let x = Color.red;
-let y = Named["string name"];
+// ADT enum construction
+const m1 = Message.Move{x: 10, y: 20};
+const m2 = Message.Write("hello world");
+
+// ADT enum deconstruction via extractors
+const Message.Move{x, y} = m1;
+const Message.Write(text) = m2;
+
+// ADT enum pattern matching
+match (m1) {
+  when Message.Move{x: let x, y: let y}: ...,
+  when Message.Write(let text): ...,
+  when Message.Quit: ...;
+}
+```
+
+## Object Literal-like Construction
+
+ADT enums support an object literal-like construction syntax:
+
+```js
+const msg = Message.Move{ x: 10, y: 20 };
 ```
 
 # Semantics
 
 ## Well-Known Symbols
 
-This proposal introduces three new well-known symbols that are used with enums:
+This proposal introduces four new well-known symbols that are used with enums:
 
-| Specification Name | \[\[Description]] | Value and Purpose |
-|:-|:-|:-|
-| @@toEnum | `"Symbol.toEnum"` | A method that is used to derive the value for an enum member during _EnumMember_ evaluation. |
-| @@formatEnum | `"Symbol.formatEnum"` | A method of an _enum object_ that is used to convert a value into a string representation based on the member names of the enum. Called by `Enum.format`. |
-| @@parseEnum | `"Symbol.parseEnum"` | A method of an _enum object_ that is used to convert a member name String into the value represented by that member of the enum. Called by `Enum.parse`. |
+| Specification Name  | \[\[Description]]            | Value and Purpose |
+|:--------------------|:-----------------------------|:------------------|
+| @@toEnum            | `"Symbol.toEnum"`            | A method that is used to derive the value for an enum member during _EnumMember_ evaluation. |
+| @@formatEnum        | `"Symbol.formatEnum"`        | A method of an _enum object_ that is used to convert a value into a string representation based on the member names of the enum. Called by `Enum.format`. |
+| @@parseEnum         | `"Symbol.parseEnum"`         | A method of an _enum object_ that is used to convert a member name String into the value represented by that member of the enum. Called by `Enum.parse`. |
+| @@propertyConstruct | `"Symbol.propertyConstruct"` | A method of an ADT _enum member_ that is used to construct an ADT record-like enum value using object literal-like syntax. |
```

Comment:
I'd like to reduce the number of new well-known symbols. Is it possible to merge them into one?

`@@enum: { format?, parse?, construct?, from? }`

Reply:
Maybe parse and format, but not the others:

- `@@toEnum` goes on the mapper passed to `of` such as `Number`, `String`, etc.
- `@@formatEnum` and `@@parseEnum` go on the constructor produced by the enum declaration_.
- `@@propertyConstruct` goes on an ADT enum tagged record member.

Those are three wholly different placements. `@@formatEnum` and `@@parseEnum` provide a way to access the underlying enum data used to relate a key to a value and vice versa, and are symbols rather than internal slots so that existing enum-like constructs can be adapted to work with `Enum` and can be produced by a transpiler.